packages/core/src/storage/compare.ts

```diff
@@ -410,6 +410,9 @@ function enumSize(memberCount: number): number {
 
 export function stripContractSubstrings(label?: string) {
   if (label !== undefined) {
-    return label.replace(/\bcontract /g, '');
+    return label
+      .replace(/\bcontract /g, '')
+      .replace(/\bstruct /g, '')
+      .replace(/\benum /g, '');
```

Comment:
Clearer intention IMO:

```suggestion
    return label.replace(/\b(contract|struct|enum) /g, '');
```